### PR TITLE
SWARM-1792: skipping cache activation for unavailable caches

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
- [X] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [X] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [X] Have you built the project locally prior to submission with `mvn clean install`?

-----
Motivation
----------
To activate the appropriate caches but get rid of error message complaining about the failed services when some caches are not available.

Modifications
-------------
`InfinispanCustomizer` and `CacheActivator` where modified to skip activating caches which are not available (their fractions are not added).
Also, undertow cache was renamed to web and jpa to hibernate. I believe these are the proper names of the cache containers.

Result
------
The error message is no longer printed in the log, yet, when possible, the required caches are started

### CODE REVIEW

File: fractions/wildfly/infinispan/src/main/java/org/wildfly/swarm/infinispan/InfinispanMessages.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * Date: 3/15/18
+ */
+@MessageLogger(projectCode = "WFSISPN", length = 4)
+public interface InfinispanMessages extends BasicLogger {
+
+    InfinispanMessages MESSAGES = Logger.getMessageLogger(InfinispanMessages.class, "org.wildfly.swarm.infinispan");
+
+    @LogMessage(level = Logger.Level.INFO)
+    @Message(id = 1, value = "Skipping activating cache configuration service: %s. The service is missing.")
+    void skippingCacheActivation(String key);
```
Comment (Ladicek) [inline fractions/wildfly/infinispan/src/main/java/org/wildfly/swarm/infinispan/InfinispanMessages.java]: The entire PR looks very good to me, the only thing I wonder about is this log message. Should it really be on INFO? Should it say "the service is missing"? (I'm thinking it should rather be DEBUG and read "the corresponding fraction is not present".)

Comment (michalszynkiewicz) [reply fractions/wildfly/infinispan/src/main/java/org/wildfly/swarm/infinispan/InfinispanMessages.java]: done

**Comment by kenfinnigan** [comment]
Another approach would be to not amend `CacheActivator` and simply not call new if it's unsatisfied. Might be a bit simpler

**Comment by michalszynkiewicz** [comment]
@kenfinnigan updated, I also squashed the commits not to put stuff into `CacheActivator` history

File: core/container/src/main/java/org/wildfly/swarm/container/runtime/RuntimeServer.java [hunk 2]
```diff
@@ -198,7 +200,9 @@ public ExtensibleConfigurationPersister createConfigurationPersister(ServerEnvir
 
         List<ServiceActivator> activators = new ArrayList<>();
 
-        this.serviceActivators.forEach(activators::add);
+        StreamSupport.stream(serviceActivators.spliterator(), false)
```
Comment (kenfinnigan) [inline core/container/src/main/java/org/wildfly/swarm/container/runtime/RuntimeServer.java:203]: Any particular reason for this usage vs `this.serviceActivators.stream()` ?

Comment (michalszynkiewicz) [reply core/container/src/main/java/org/wildfly/swarm/container/runtime/RuntimeServer.java:203]: `Instance::stream` is not available in CDI 1.2, it was added in 2.0 https://docs.jboss.org/cdi/api/2.0/javax/enterprise/inject/Instance.html#stream--

Comment (kenfinnigan) [reply core/container/src/main/java/org/wildfly/swarm/container/runtime/RuntimeServer.java:203]: Ah, I think I got confused with the `List` just above that code with a similar name. Forgot it was dealing with `Instance`.

Ignore me

**Comment by gastaldi** [comment]
I can confirm that this PR solves the SWARM-1792 issue. Good work @michalszynkiewicz ! :beers:

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**